addons/membership/membership.py

```diff
@@ -136,6 +137,11 @@ def _state(self, cr, uid, ids, name, args, context=None):
             res[line.id] = state
         return res
 
+    def _set_state(self, cr, uid, ids, field_name, field_value, args, context=None):
+        cr.execute('UPDATE membership_membership_line '\
+                       'SET state=%s '\
+                       'WHERE id IN %s', (field_value, tuple([ids]),))
+        return True
```

Comment:
Currently in master there are no any functionality to know genrated refund for which invoice it is (means there are no any link to know).so,we can't find refund so it is not possible any other way.
You can see this task: https://www.odoo.com/web?#id=11398&view_type=form&model=project.task&action=327